### ISSUE TITLE
Check-in generated Go files

### ISSUE THREAD

**Issue by johnnyaug** [treeverse/lakeFS#5758]
## Reasoning
1. No need to generate the code when checking out the repo.
1. We will be able to remove annoying code generation steps from the build and tests.
1. Other types of generated code are already checked-in, so it's not like our repository is "normalized".
1. Our Go SDK becomes `go get`-able.

## DoD
1. Generated Go files are checked in.
3. Tests do not pass if these files are not up to date.
4. "Generate code" step is removed from Esti tests.
5. Any documentation explaining how/when to generate code reflects the new situation.

**Comment by arielshaqed** [comment]
## tl;dr:

I am split on this.  I think we should list pros and cons, it's non-trivial.  It makes life easier for casual readers, it probably makes test actions run faster.  It makes life for some developers (for instance, this one) harder.

## +++

Most of the above:

* Easier to checkout and Go
* Slightly faster build

And we probably also get:

* Improved vulnerability scanning
* Slightly improved semantic linking (click symbol to go to its definition)
* `webui/` is already _not_ autogenerated
* Faster test actions!  This step _can be measured_ (hint hint) ⏱️ 

## 000

* We're halfway there (_Other types of generated code are already checked-in, so it's not like our repository is "normalized"._): so not an argument in favour of going either way. :-/

## ---

* Harder to push code: We already have missing "make gen" calls on pushed codes.  These fail checks-validator during our checks action, and need to be re-pushed after a few minutes.
* Conflicts from Hell when generated code on trunk has changed since branching out to a dev branch.  checks-validator makes these less dangerous to trunk, but don't help programmers remain sane.
* `webui/` is already _not_ autogenerated, **BUT** we cannot check it with checks-validator!
* Increased confusion with the Python and Java client codes: these _are_ auto-generated but because of the way non-Go dependencies work any using code does _not_ see the generated code.
* Denormalization makes it more likely to have inconsistency in built code!  We now rely on an _action_ running successfully rather than on a complete build

**Comment by ozkatz** [comment]
Adding a possible pro and a possible con:

Pro: This should make lakeFS `go get`-able, so we also win a Go SDK?

Con: To avoid inconsistencies, you'd still want to make sure there's no diff between generated code and whatever is checked in.. this might reduce any potential gain in build / test speed?
